covid-projections/act-now-packages
# Make React and React-dom peer dependencies

Close https://github.com/covid-projections/act-now-packages/issues/93

I think we need to also add `react`/`react-dom` as devDependencies to use storybook, from https://github.com/storybookjs/storybook/issues/13665.

**To test/verify:** 
1. Create a test project with `@actnowcoalition/ui-components` as a dependency, but without `react`/`react-dom` as dependencies. E.g, a `package.json` like:
```json
{
  "name": "peer-test",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@actnowcoalition/ui-components": "file:../act-now-packages/packages/ui-components"
  },
  "devDependencies": {
    "@types/node": "18.7.18"
  }
}
```
2. Use the local/un-published `@actnowcoalition/ui-components` changes by building the package and using `yarn link` or accessing the source files directly in the test project. 
3. run `yarn` in the test project
4. Verify that `react` and `react-dom` are not present in the test project `yarn.lock` file, and check that there `unmet peer dependency` warnings in the output of step 3 (see below).
<img width="997" alt="image" src="https://user-images.githubusercontent.com/55333380/190445483-bfcecb9d-f31f-4ecd-baa4-d4c9efb61530.png">
